### PR TITLE
Feature - Implement stop function

### LINKED ISSUE #25: Implement stop() function
**Describe the new feature you would like to see**
Implement the stop() function of the robot interface

**Describe the solution you'd like**
Should use GraphQL

**How will this feature affect the current Threat Model?**
N/A

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #25

### CODE REVIEW

File: src/isar_exr/api/energy_robotics_api.py [hunk 1]
```diff
@@ -39,3 +42,26 @@ def is_mission_running(self, exr_robot_id: str):
         response_dict: dict[str, Any] = self.client.query(query_string, params)
         is_running: bool = response_dict["isMissionRunning"]
         return is_running
+
+    def pause_current_mission(self, exr_robot_id: str) -> None:
+        query_string: str = """
+            mutation pauseMission($robot_id: String!)
+                {
+                    pauseMissionExecution(robotID:$robot_id)
```
Comment (eivindsjovold) [inline src/isar_exr/api/energy_robotics_api.py:50]: Can you test that this function raises and returns as expected?

File: src/isar_exr/robotinterface.py [hunk 2]
```diff
@@ -58,23 +64,23 @@ def __init__(self):
...
 
     def step_status(self) -> StepStatus:
         return StepStatus.Successful
 
-    def stop(self) -> bool:
-        return True
+    def stop(self) -> None:
+        try:
+            self.api.pause_current_mission(self.exr_robot_id)
+        except RobotException:
```
Comment (eivindsjovold) [inline src/isar_exr/robotinterface.py:77]: or test this one

File: test.py [hunk 3]
```diff
@@ -1,31 +1,6 @@
...
-        print()
-        print(result)
-    except GraphQLError as e:
-        print(f"Error during query: {e.message}")
-    except TransportError as e:
-        print(f"Error during communcation with server: {e}")
-
-    sleep(2)
+# Test functions here:
+robot.stop()
```
Comment (eivindsjovold) [inline test.py:6]: Tests with asserts instead?

**Review by eivindsjovold — CHANGES_REQUESTED** [review]

File: test.py (hunk 3 above)
Comment (GodVenn) [reply test.py:6]: No, test.py is not a test file

File: src/isar_exr/api/energy_robotics_api.py (hunk 1 above)
Comment (GodVenn) [reply src/isar_exr/api/energy_robotics_api.py:50]: Agree fixed

File: src/isar_exr/robotinterface.py (hunk 2 above)
Comment (GodVenn) [reply src/isar_exr/robotinterface.py:77]: Agree, fixed

File: src/isar_exr/api/energy_robotics_api.py [hunk 4]
```diff
@@ -39,3 +42,26 @@ def is_mission_running(self, exr_robot_id: str):
...
+                    }
+                }
+        """
+        params: dict = {"robot_id": exr_robot_id}
+        try:
+            result: Dict[str, Any] = self.client.query(query_string, params)
+        except Exception as e:
+            raise RobotException(e)
+        success: bool = (
+            result["status"] == "PAUSE_REQUESTED" or result["status"] == "PAUSED"
```
Comment (eivindsjovold) [inline src/isar_exr/api/energy_robotics_api.py:62]: Call function to convert to step status? get rid of or and string based

File: tests/api/test_energy_robotics_api.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    wrong_response: Dict[str, Any] = {"status": "ANYTHING_ELSE"}
+
+    @mock.patch.object(Client, "execute", mock.Mock(return_value=paused_response))
+    def test_succeeds_if_status_is_paused(self):
+        api: EnergyRoboticsApi = EnergyRoboticsApi()
+        api.pause_current_mission("test_exr_robot_id")
+
+    @mock.patch.object(
+        Client, "execute", mock.Mock(return_value=pause_requested_response)
+    )
```
Comment (eivindsjovold) [inline tests/api/test_energy_robotics_api.py:26]: is there a test objective here?

File: tests/api/test_energy_robotics_api.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    def test_succeeds_if_status_is_paused_requested(self):
+        api: EnergyRoboticsApi = EnergyRoboticsApi()
+        api.pause_current_mission("test_exr_robot_id")
```
Comment (eivindsjovold) [inline tests/api/test_energy_robotics_api.py:29]: objective?

**Review by eivindsjovold — CHANGES_REQUESTED** [review]

File: src/isar_exr/api/energy_robotics_api.py (hunk 4 above)
Comment (GodVenn) [reply src/isar_exr/api/energy_robotics_api.py]: Fixed

**Review by eivindsjovold — APPROVED** [review]
LGTM when linting ok

**Pull request merged**